includes/class-newspack-newsletters.php

```diff
@@ -1176,17 +1174,17 @@ public static function is_newsletter_sent( $post_id ) {
 		}
 
 		/** Detect meta that determines the sent state */
-		$sent = get_post_meta( $post_id, 'newsletter_sent', true );
-		if ( 0 < $sent ) {
+		$sent         = get_post_meta( $post_id, 'newsletter_sent', true );
+		$post_status  = get_post_status( $post_id );
+		$is_published = 'publish' === $post_status || 'private' === $post_status;
+		$publish_date = $is_published ? get_post_datetime( $post_id, 'date', 'gmt' )->getTimestamp() : 0;
+		if ( 0 < $sent && $sent === $publish_date ) {
 			return $sent;
 		}
 
-		/** Legacy check for sent/publish newsletters without meta. */
-		if ( 'publish' === get_post_status( $post_id ) ) {
-			$post = get_post( $post_id );
-			$sent = strtotime( $post->post_date );
-			self::set_newsletter_sent( $post_id, $sent );
-			return $sent;
+		if ( $publish_date ) {
```

Comment:
It's odd that we're using the $publish_date to determine whether the newsletter should be marked as sent. Why does this PR change the original behavior?

Reply:
It's because for any sites that exhibit the bug described in the PR description (where the `newsletter_sent` timestamp stored in meta doesn't match the actual sent time due to the timezone difference) we want to overwrite the existing `newsletter_sent` meta with the actual timezone-correct timestamp. To walk through the code step-by-step:

1. `$sent         = get_post_meta( $post_id, 'newsletter_sent', true );` - we check if the already has a sent timestamp saved as post meta
2. `$post_status  = get_post_status( $post_id );` and `$is_published = 'publish' === $post_status || 'private' === $post_status;` - this uses the post status to determine whether the newsletter was sent
3. `$publish_date = $is_published ? get_post_datetime( $post_id, 'date', 'gmt' )->getTimestamp() : 0;` - if it was published, we get the publish date as a timestamp using [`get_post_datetime`](https://developer.wordpress.org/reference/functions/get_post_datetime/) (which returns a `DateImmutable` object and which will return a timestamp correctly calculated based on the site's timezone setting)
4. `if ( 0 < $sent && $sent === $publish_date ) { return $sent; }` - if the timestamp saved to post meta already matches the publish date, then we're good—just return that
5. If not, then we have the correct publish date form step 3, so save that as the `newsletter_sent` meta value for the next time.

As always I'm open to improvements if you have ideas on how to accomplish this better!

Reply:
Ah, I see! Sorry for my confusion!